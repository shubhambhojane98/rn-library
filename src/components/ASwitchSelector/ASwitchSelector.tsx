import React, {FC, useState} from 'react';
import {StyleSheet, View, TouchableOpacity} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {Color} from '../../theme';
import ATypography from '../ATypography/ATypography';
import {defaultScale} from '../../utils/Common';

interface Props {
  option1: string;
  option2: string;
  onSelectSwitch: any;
  width?: number;
}

const ASwitchSelector: FC<Props> = ({
  option1,
  option2,
  onSelectSwitch,
  width,
}) => {
  const [getSelectionMode, setSelectionMode] = useState(2);

  const toggleSwitch = val => {
    setSelectionMode(val);
    onSelectSwitch(val);
  };

  return (
    <View style={{...styles.switchContainer, width: width}}>
      <TouchableOpacity
        activeOpacity={1}
        onPress={() => toggleSwitch(1)}
        style={{
          ...styles.switchBtn,
          backgroundColor:
            getSelectionMode === 1 ? Color.yellow : Color.switchBackgroundColor,
        }}>
        <ATypography
          children={option1}
          fontSize={16}
          variant={'primaryBold'}
          style={{...styles.text}}
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={1}
        onPress={() => toggleSwitch(2)}
        style={{
          ...styles.switchBtn,
          backgroundColor:
            getSelectionMode === 2 ? Color.yellow : Color.switchBackgroundColor,
        }}>
        <ATypography
          children={option2}
          fontSize={16}
          variant={'primaryBold'}
          style={{...styles.text}}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  switchContainer: {
    height: moderateScale(52, defaultScale),
    backgroundColor: Color.switchBackgroundColor,
    borderRadius: 25,
    flexDirection: 'row',
    justifyContent: 'center',
    padding: moderateScale(2, defaultScale),
  },
  switchBtn: {
    flex: 1,
    margin: moderateScale(5, defaultScale),
    width: moderateScale(109, defaultScale),
    height: moderateScale(40, defaultScale),
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: Color.textColor,
  },
});

export default ASwitchSelector;
