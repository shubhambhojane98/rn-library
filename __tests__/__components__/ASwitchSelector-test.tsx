import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ASwitchSelector from '../../src/components/ASwitchSelector/ASwitchSelector';

it('renders correctly default prop', () => {
  const tree = renderer
    .create(
      <ASwitchSelector option1="one" option2="2" onSelectSwitch={jest.fn} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with width prop', () => {
  const tree = renderer
    .create(
      <ASwitchSelector
        option1="one"
        option2="2"
        width={200}
        onSelectSwitch={jest.fn}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
