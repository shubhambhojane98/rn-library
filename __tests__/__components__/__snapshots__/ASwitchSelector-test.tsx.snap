// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default prop 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D7D7D7",
      "borderRadius": 25,
      "flexDirection": "row",
      "height": 63.885714285714286,
      "justifyContent": "center",
      "padding": 2.4571428571428573,
      "width": undefined,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D7D7D7",
        "borderRadius": 25,
        "flex": 1,
        "height": 49.14285714285714,
        "justifyContent": "center",
        "margin": 6.142857142857142,
        "opacity": 1,
        "width": 133.9142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 19.65714285714286,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#231F20",
          },
        ]
      }
    >
      one
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAD000",
        "borderRadius": 25,
        "flex": 1,
        "height": 49.14285714285714,
        "justifyContent": "center",
        "margin": 6.142857142857142,
        "opacity": 1,
        "width": 133.9142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 19.65714285714286,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#231F20",
          },
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`renders correctly with width prop 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D7D7D7",
      "borderRadius": 25,
      "flexDirection": "row",
      "height": 63.885714285714286,
      "justifyContent": "center",
      "padding": 2.4571428571428573,
      "width": 200,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D7D7D7",
        "borderRadius": 25,
        "flex": 1,
        "height": 49.14285714285714,
        "justifyContent": "center",
        "margin": 6.142857142857142,
        "opacity": 1,
        "width": 133.9142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 19.65714285714286,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#231F20",
          },
        ]
      }
    >
      one
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAD000",
        "borderRadius": 25,
        "flex": 1,
        "height": 49.14285714285714,
        "justifyContent": "center",
        "margin": 6.142857142857142,
        "opacity": 1,
        "width": 133.9142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 19.65714285714286,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#231F20",
          },
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;
